src/vm/core.zig

```diff
@@ -358,6 +358,56 @@ pub const CairoVM = struct {
         }
     }
 
+    /// Attempts to deduce `op1` and `res` for an instruction, given `dst` and `op0`.
+    ///
+    /// # Arguments
+    ///
+    /// - `inst`: The instruction to deduce `op1` and `res` for.
+    ///
+    /// - `dst`: The destination of the instruction.
+    ///
+    /// - `op0`: The first operand of the instruction.
+    ///
+    /// # Returns
+    ///
+    /// - `Tuple`: A tuple containing the deduced `op1` and `res`.
+    fn deduce_op1(self: *const CairoVM, inst: *const instructions.Instruction, dst: ?*const relocatable.MaybeRelocatable, op0: ?*const relocatable.MaybeRelocatable) !std.meta.Tuple(&[_]type{ ?relocatable.MaybeRelocatable, ?relocatable.MaybeRelocatable }) {
+        _ = self;
```

Comment:
Would be good to have that as a function and not as a method to avoid this, what do you think?

Reply:
I followed what the python implementation was doing, but I planned to ask about this as well once I'm settled with the actual implementation.

But yeah, this should probably not be a member.

Do I just move it out of the struct definition?

Reply:
Actually, I just saw other funcitons of the same kind outside of the struct. That makes much sense x)